pysatNASA/instruments/methods/_cdf.py

```diff
@@ -0,0 +1,442 @@
+    datetime : bool
+        True if epoch dependency data should be in datetime format
+        (default=True)
+
+    """
+
+    def __init__(self, filename,
+                 varformat='*',  # Regular expressions
+                 var_types=['data', 'support_data'],
+                 center_measurement=False,
+                 raise_errors=False,
+                 regnames=None,
+                 datetime=True,
+                 **kwargs):
+        self._raise_errors = raise_errors
+        self._filename = filename
+        self._varformat = varformat
+        self._var_types = var_types
+        self._datetime = datetime
+        self._center_measurement = center_measurement
+
+        # Registration names map from file params to kamodo-compatible names
+        if regnames is None:
+            regnames = {}
+        self._regnames = regnames
+
+        self._cdf_file = cdflib.CDF(self._filename)
+        self._cdf_info = self._cdf_file.cdf_info()
+        self.data = {}
+        self.meta = {}
+        self._dependencies = {}
+
+        self._variable_names = (self._cdf_info['rVariables']
+                                + self._cdf_info['zVariables'])
+
+        self.load_variables()
+
+    def __enter__(self):
+        """Context manager protocol
+        enters runtime context related to this object.
+        The with statement will bind this method’s return value
+        to the target(s) specified in the as clause of the statement
+        """
+        return self
+
+    def __exit__(self, type, value, tb):
+        """Context manager protocol
+        If any exceptions occur while attempting to execute the block
+        of code nested after the with statement, Python will pass
+        information about the exception into this method
+
+        Parameters
+        ----------
+        type : str
+            type of exception
+        value : str
+            value of exception
+        tb : traceback
+            traceback object generated by exception
+        """
+        pass
+
+    def get_dependency(self, x_axis_var):
+        """Retrieves variable dependency unique to filename
+
+        Parameter
+        ---------
+        x_axis_var : string
+            name of variable
+        """
+
+        return self._dependencies.get(self._filename + x_axis_var)
+
+    def set_dependency(self, x_axis_var, x_axis_data):
+        """Sets variable dependency unique to filename
+
+        Parameters
+        ----------
+        x_axis_var : str
+            name of variable
+        x_axis_data : array-like
+            epoch dependency data
+        """
+
+        self._dependencies[self._filename + x_axis_var] = x_axis_data
+
+    def set_epoch(self, x_axis_var):
+        """Stores epoch dependency
+
+        Parameters
+        ----------
+        x_axis_var : str
+            name of variable
+
+        """
+
+        data_type_description \
+            = self._cdf_file.varinq(x_axis_var)['Data_Type_Description']
```

Comment:
Recommend breaking this on the open paren.
```suggestion
        data_type_description = self._cdf_file.varinq(
            x_axis_var)['Data_Type_Description']
```